### PR TITLE
Raft message log line rewrite to aid machine readability.

### PR DESCRIPTION
**Pull request by andrewkerr9000**
Include date, both to and from members, type of message rather than class of message, arranged as space separated fields and with message quoted.

### CODE REVIEW

**Comment by andrewkerr9000** [comment]
Example logs from a test. For a given node the first MemberId will always be that node's. It's useful to have in tests or when combining logs from multiple nodes, and it'll compress well.

```MemberId{690df055} --- MemberId{690df055} Info "I am MemberId{690df055}"
2017-10-09 11:01:07.834 MemberId{62995260} --- MemberId{62995260} Info "I am MemberId{62995260}"
2017-10-09 11:01:07.834 MemberId{68176ac2} --- MemberId{68176ac2} Info "I am MemberId{68176ac2}"
2017-10-09 11:01:07.854 MemberId{690df055} --> MemberId{68176ac2} VOTE_REQUEST "Vote.Request from MemberId{690df055} {term=1, candidate=MemberId{690df055}, lastAppended=0, lastLogTerm=0}"
2017-10-09 11:01:07.854 MemberId{690df055} --> MemberId{62995260} VOTE_REQUEST "Vote.Request from MemberId{690df055} {term=1, candidate=MemberId{690df055}, lastAppended=0, lastLogTerm=0}"
2017-10-09 11:01:07.855 MemberId{68176ac2} <-- MemberId{690df055} VOTE_REQUEST "Vote.Request from MemberId{690df055} {term=1, candidate=MemberId{690df055}, lastAppended=0, lastLogTerm=0}"
2017-10-09 11:01:07.857 MemberId{68176ac2} --> MemberId{690df055} VOTE_RESPONSE "Vote.Response from MemberId{68176ac2} {term=1, voteGranted=true}"
2017-10-09 11:01:07.857 MemberId{62995260} <-- MemberId{690df055} VOTE_REQUEST "Vote.Request from MemberId{690df055} {term=1, candidate=MemberId{690df055}, lastAppended=0, lastLogTerm=0}"
2017-10-09 11:01:07.858 MemberId{62995260} --> MemberId{690df055} VOTE_RESPONSE "Vote.Response from MemberId{62995260} {term=1, voteGranted=true}"
2017-10-09 11:01:07.858 MemberId{690df055} <-- MemberId{68176ac2} VOTE_RESPONSE "Vote.Response from MemberId{68176ac2} {term=1, voteGranted=true}"
2017-10-09 11:01:07.863 MemberId{690df055} --> MemberId{68176ac2} HEARTBEAT "Heartbeat from MemberId{690df055} {leaderTerm=1, commitIndex=-1, commitIndexTerm=-1}"
2017-10-09 11:01:07.863 MemberId{690df055} --> MemberId{62995260} HEARTBEAT "Heartbeat from MemberId{690df055} {leaderTerm=1, commitIndex=-1, commitIndexTerm=-1}"
2017-10-09 11:01:07.891 MemberId{690df055} --> MemberId{68176ac2} APPEND_ENTRIES_REQUEST "AppendEntries.Request from MemberId{690df055} {leaderTerm=1, prevLogIndex=0, prevLogTerm=0, entry=[], leaderCommit=-1}"
2017-10-09 11:01:07.893 MemberId{690df055} --> MemberId{62995260} APPEND_ENTRIES_REQUEST "AppendEntries.Request from MemberId{690df055} {leaderTerm=1, prevLogIndex=0, prevLogTerm=0, entry=[], leaderCommit=-1}"
2017-10-09 11:01:07.894 MemberId{68176ac2} <-- MemberId{690df055} HEARTBEAT "Heartbeat from MemberId{690df055} {leaderTerm=1, commitIndex=-1, commitIndexTerm=-1}"
2017-10-09 11:01:07.895 MemberId{68176ac2} --> MemberId{690df055} HEARTBEAT_RESPONSE "HeartbeatResponse{from=MemberId{68176ac2}}"
2017-10-09 11:01:07.895 MemberId{62995260} <-- MemberId{690df055} HEARTBEAT "Heartbeat from MemberId{690df055} {leaderTerm=1, commitIndex=-1, commitIndexTerm=-1}"
2017-10-09 11:01:07.895 MemberId{62995260} --> MemberId{690df055} HEARTBEAT_RESPONSE "HeartbeatResponse{from=MemberId{62995260}}"```

**Review by phughk — APPROVED** [review]

File: enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/logging/BetterMessageLogger.java [hunk 1]
```diff
@@ -20,49 +20,69 @@
...
+
+        Direction( String arrow )
+        {
+            this.arrow = arrow;
+        }
+    }
+
     private final PrintWriter printWriter;
-    private DateFormat dateFormat = new SimpleDateFormat("HH:mm:ss.SSS");
+    private final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern( "yyyy-MM-dd HH:mm:ss.SSS" );
```
Comment (digitalstain) [inline enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/logging/BetterMessageLogger.java]: This does not include timezone information, correct? This information is often useful so we can correlate between the messages in this log and debug/neo4j logs.

Comment (andrewkerr9000) [reply enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/logging/BetterMessageLogger.java]: Fixed

Comment (andrewkerr9000) [reply enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/logging/BetterMessageLogger.java]: Updated to include time zone provided by clock. We don't seem to have a common clock in the Modules, I used `Clocks.systemClock`

**Comment by andrewkerr9000** [comment]
Updated date time format.

```
2017-10-11 13:05:51.112+0000 MemberId{6b07935b} --- MemberId{6b07935b} Info "I am MemberId{6b07935b}"
2017-10-11 13:05:51.144+0000 MemberId{113862dd} --- MemberId{113862dd} Info "I am MemberId{113862dd}"
2017-10-11 13:05:51.145+0000 MemberId{29527bf7} --- MemberId{29527bf7} Info "I am MemberId{29527bf7}"
2017-10-11 13:05:51.161+0000 MemberId{6b07935b} --> MemberId{113862dd} VOTE_REQUEST "Vote.Request from MemberId{6b07935b} {term=1, candidate=MemberId{6b07935b}, lastAppended=0, lastLogTerm=0}"
2017-10-11 13:05:51.161+0000 MemberId{6b07935b} --> MemberId{29527bf7} VOTE_REQUEST "Vote.Request from MemberId{6b07935b} {term=1, candidate=MemberId{6b07935b}, lastAppended=0, lastLogTerm=0}"
2017-10-11 13:05:51.162+0000 MemberId{113862dd} <-- MemberId{6b07935b} VOTE_REQUEST "Vote.Request from MemberId{6b07935b} {term=1, candidate=MemberId{6b07935b}, lastAppended=0, lastLogTerm=0}"
2017-10-11 13:05:51.163+0000 MemberId{113862dd} --> MemberId{6b07935b} VOTE_RESPONSE "Vote.Response from MemberId{113862dd} {term=1, voteGranted=true}"
2017-10-11 13:05:51.164+0000 MemberId{29527bf7} <-- MemberId{6b07935b} VOTE_REQUEST "Vote.Request from MemberId{6b07935b} {term=1, candidate=MemberId{6b07935b}, lastAppended=0, lastLogTerm=0}"
2017-10-11 13:05:51.164+0000 MemberId{29527bf7} --> MemberId{6b07935b} VOTE_RESPONSE "Vote.Response from MemberId{29527bf7} {term=1, voteGranted=true}"
2017-10-11 13:05:51.165+0000 MemberId{6b07935b} <-- MemberId{113862dd} VOTE_RESPONSE "Vote.Response from MemberId{113862dd} {term=1, voteGranted=true}"
2017-10-11 13:05:51.169+0000 MemberId{6b07935b} --> MemberId{113862dd} HEARTBEAT "Heartbeat from MemberId{6b07935b} {leaderTerm=1, commitIndex=-1, commitIndexTerm=-1}"
```

File: enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/logging/BetterMessageLogger.java (hunk 1 above)
Comment (MishaDemianenko) [reply enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/logging/BetterMessageLogger.java]: we have PlatformModule::clock thingy

Comment (andrewkerr9000) [reply enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/logging/BetterMessageLogger.java]: Not in 3.1. It's also not used for logging, which afaics is created via a builder specifying UTC timezone.

**Pull request merged**